source/docs/software/commandbased/commands-v3/migration-from-v2.rst

```diff
@@ -0,0 +1,112 @@
+# Migrating from Commands v2 to v3
+
+.. todo::
+   This article guides teams through migrating an existing Commands v2 codebase to v3. This article should help teams plan their migration strategy, understand the key differences, and avoid common pitfalls. This is a critical resource for established teams.
+
+## Migration Strategy
+
+.. todo::
+   Provide high-level guidance on planning a v2-to-v3 migration. Cover:
+   - Should teams migrate? (pros/cons, when to migrate vs. stay on v2)
+   - Migration approach: full rewrite recommended
+   - What to migrate first (start with simple mechanisms/commands)
+   - Testing strategy during migration
+   - Timeline considerations (offseason vs. competition season)
+   - Team training and learning curve
+   - Recommended migration phases
+
+## Package Changes
+
+.. todo::
+   Document package and import changes between v2 and v3. Cover:
+   - New v3 package names and locations
+   - Renamed classes and methods
+   - Deprecated v2 APIs and their v3 equivalents
+   - How to add v3 dependencies to build.gradle/pom.xml
+   - Version compatibility and requirements
+   - Note: Vendor library compatibility with v3 is up to individual vendors
+
+## History: Evolution of Command-Based Programming
+
+.. todo::
+   Provide context on how the command-based framework has evolved. Cover:
+   - Brief history: v1 (original command-based) → v2 (functional composition) → v3 (coroutines)
+   - Why each version was created and what problems it solved
+   - How v3 addresses the learning curve issues with v2's declarative composition
+
+## Concept Mapping
```

Comment:
This feels like it should be higher in the TOC before talking about the migration strategy